crabs/detection_tracking/datamodules.py

```diff
@@ -163,18 +166,27 @@ def setup(self, stage: str):
         """
 
         # Assign transforms
+        # right now assuming validation and test get the same transform
         self.train_transform = self._get_train_transform()
         self.test_transform = self._get_test_val_transform()
         self.val_transform = self._get_test_val_transform()
 
-        # Assign datasets for dataloader depending on stage
+        # Assign datasets for dataloader depending on stage.
         # omitting "predict" stage for now
-        train_dataset, test_dataset, val_dataset = self._compute_splits()
         if stage == "fit":
+            # compute splits passing the relevant transform
```

Comment:
we don't need this condition actually. Lightning will deal with it.
after assigning the transform..
```
self.train_dataset, _, _ = self._compute_splits(self.train_transform)
_, _, self.val_dataset = self._compute_splits(self.val_transform)
_, self.test_dataset, _ = self._compute_splits(self.test_transform)
```

Reply:
I couldn't find this update in the docs - would you mind linking to it? 

So without the stage we now compute all splits, regardless of the stage. Is that correct?